framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2401,8 +2423,31 @@ public boolean isSupportedQualifier(@Nullable AnnotationMirror a) {
         if (a == null) {
             return false;
         }
-        return AnnotationUtils.containsSameByName(
-                this.getQualifierHierarchy().getTypeQualifiers(), a);
+        return isSupportedQualifier(AnnotationUtils.annotationName(a));
+    }
+
+    /**
+     * Determines whether the given class is a part of the type system under which this type factory
+     * operates.
+     *
+     * @param className annotation class
+     * @return true if that class is a type qualifier in the type system under which this type
+     *     factory operates, false otherwise
+     */
+    public boolean isSupportedQualifier(Class<? extends Annotation> className) {
+        return isSupportedQualifier(className.getCanonicalName());
```

Comment:
I like the change to remove `AnnotationMirrors` for the supported type qualifiers.
I'm not so sure why we also need the new String versions of these APIs.
Here I'm also confused why this uses the new set of Strings instead of the existing `supportedQuals`. This seems to introduce a duplication between `supportedQuals` and `supportedQualsNames` and it's not clear to me why we need it and what is good about it.
Could you expand on the motivation for this a bit?

Reply:
We don't *need* the String version of this method.  We do need `supportedQualsNames` so that we can compare the String name against the AnnotationMirror name for `isSupportedQualifier(AnnotationMirror)`.

Reply:
That didn't answer why this isn't just `return getSupportedTypeQualifiers().contains(className)`. The two sets contain corresponding values, so why query the String set instead of the Class set?